packages/opentelemetry-core/src/context/propagation/B3Propagator.ts

```diff
@@ -71,29 +88,48 @@ export class B3Propagator implements HttpTextPropagator {
   extract(context: Context, carrier: unknown, getter: GetterFunction): Context {
     const traceIdHeader = getter(carrier, X_B3_TRACE_ID);
     const spanIdHeader = getter(carrier, X_B3_SPAN_ID);
+    const parentSpanIdHeader = getter(carrier, X_B3_PARENT_SPAN_ID);
     const sampledHeader = getter(carrier, X_B3_SAMPLED);
+    const flagsHeader = getter(carrier, X_B3_FLAGS);
 
     const traceIdHeaderValue = Array.isArray(traceIdHeader)
       ? traceIdHeader[0]
       : traceIdHeader;
     const spanId = Array.isArray(spanIdHeader) ? spanIdHeader[0] : spanIdHeader;
+    const parentSpanId = Array.isArray(parentSpanIdHeader)
```

Comment:
Nit: (DRY) - because this pattern is repeated can something be done about it ?
```
const var = Array.isArray( ......
```
maybe a helper function and then
```
const options = parseHeader(sampledHeader);
...
```
or maybe this could work ?
```
const options = sampledHeader[0] ?? sampledHeader;
```
or something else just to avoid repeating the same block pattern, WDYT ?

Reply:
sounds good. I can add a helper. The last option won't work because if the header is a string it will just return the first character.